patches/SDL-gamecontroller.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/src/video/uikit/SDL_uikitmetalview.m b/src/video/uikit/SDL_uikitmetalview.m
+index 8274eaff9..04887a0d0 100644
+--- a/src/video/uikit/SDL_uikitmetalview.m
++++ b/src/video/uikit/SDL_uikitmetalview.m
+@@ -35,7 +35,9 @@
+ 
+ #import "SDL_uikitwindow.h"
+ #import "SDL_uikitmetalview.h"
++#import <GameController/GameController.h>
+ 
++GCVirtualController *CONTROLLER = NULL;
+ 
+ @implementation SDL_uikitmetalview
+ 
+@@ -54,6 +56,25 @@ - (instancetype)initWithFrame:(CGRect)frame
+         [self updateDrawableSize];
+     }
+ 
++    if ([GCController controllers].count == 0 && ![GCKeyboard coalescedKeyboard]) {
++        if (!CONTROLLER) {
++            GCVirtualControllerConfiguration *configuration = [GCVirtualControllerConfiguration new];
```

Comment:
is there a way to do this without GCVirtualController? my only iOS device is on iOS 14.2 so i can't test this :(

Reply:
I don't know of another way to add a custom controller to the GCController system.

Reply:
thats fair, can you demo the touch controls though since i can't test them? code looks fine

Reply:
(also sorry for the late response i kind of just forgot)

Reply:
https://github.com/user-attachments/assets/0e2fcfe3-a9eb-4c83-92dd-a3918d01cfd5

Reply:
that's really sick! thanks for this - one suggestion though; would you mind wrapping the controller initialization code in something like
```objc
/// Objective-C
if (@available(iOS 15, *)) {
    // meow
}
```
as per https://developer.apple.com/documentation/xcode/running-code-on-a-specific-version ?
that'd allow for this to be a single app that can have the support on iOS 15 and above and just wont initialize it on iOS 14.x and below

Reply:
Added

Reply:
tysm!